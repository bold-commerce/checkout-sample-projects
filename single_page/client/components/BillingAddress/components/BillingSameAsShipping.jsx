import { RadioField } from '@boldcommerce/stacks-ui';
import React from 'react';
import { useTranslation } from 'react-i18next';

const BillingSameAsShipping = ({
  billingSameAsShipping,
  setBillingSameAsShipping,
  disabled,
}) => {
  const { t } = useTranslation();
  return (
  <>
    <RadioField
      label={t('billing.same_as_shipping')}
      name="billing-address"
      className="RadioButton"
      value="SAME_AS_SHIPPING_ADDRESS"
      checked={billingSameAsShipping}
      onChange={() => setBillingSameAsShipping(true)}
      disabled={disabled}
    />
    <RadioField
      label={t('billing.different_from_shipping')}
      name="billing-address"
      className="RadioButton"
      value="DIFFERENT_BILLING_ADDRESS"
      checked={!billingSameAsShipping}
      onChange={() => setBillingSameAsShipping(false)}
      disabled={disabled}
    />
  </>
  );
};

export default BillingSameAsShipping;
