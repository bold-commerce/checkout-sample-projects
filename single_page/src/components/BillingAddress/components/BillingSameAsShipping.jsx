import { RadioField } from '@boldcommerce/stacks-ui';
import React from 'react';

const BillingSameAsShipping = ({
  billingSameAsShipping,
  setBillingSameAsShipping,
  disabled,
}) => (
  <>
    <RadioField
      label="Same as shipping address"
      name="billing-address"
      className="RadioButton"
      value="SAME_AS_SHIPPING_ADDRESS"
      checked={billingSameAsShipping}
      onChange={() => setBillingSameAsShipping(true)}
      disabled={disabled}
    />
    <RadioField
      label="Use a different billing address"
      name="billing-address"
      className="RadioButton"
      value="DIFFERENT_BILLING_ADDRESS"
      checked={!billingSameAsShipping}
      onChange={() => setBillingSameAsShipping(false)}
      disabled={disabled}
    />
  </>
);

export default BillingSameAsShipping;
