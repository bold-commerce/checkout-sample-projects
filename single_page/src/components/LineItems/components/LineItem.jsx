import React from 'react';
import PropTypes from 'prop-types';
import Product from './Product';
import { Price } from '@boldcommerce/stacks-ui';
import { useTranslation } from 'react-i18next';

const LineItem = ({
  title,
  image,
  quantity,
  totalPrice,
  variants
}) => {
  const { t } = useTranslation();
  return(
    <div className="SummaryBlock CartItem" >
      <Product
        title={title}
        image={image}
        variants={variants}
      />
      <div className="CartItem__QuantityPriceWrapper">
        <div className="CartItem__ProductQuantityWrapper">
          <div className="ProductQuantity" aria-label="product quantity">
            {quantity}
          </div>
        </div>
        <div className="CartItem__ProductPrice">
          <Price amount={totalPrice} moneyFormatString={t('currency_format')} />
        </div>
      </div>
    </div>
  );}
  
  LineItem.propTypes = {
    title: PropTypes.string,
    image: PropTypes.string,
    quantity: PropTypes.number,
    lineItemKey: PropTypes.string,
    variants: PropTypes.array,
  };
  
  export default LineItem;