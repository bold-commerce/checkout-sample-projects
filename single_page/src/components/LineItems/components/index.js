export { default as LineItem } from './LineItem';
export { default as Product } from './Product';
