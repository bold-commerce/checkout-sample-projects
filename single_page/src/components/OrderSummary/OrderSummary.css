.OrderSummaryItem {
  display: flex;
  justify-content: space-between;
  font-size: 14px;
  align-items: center;
}

.OrderSummaryItem__Title {
  color: #737373;
}

.OrderSummaryItem__Amount {
  color: #3A3A3A;
}

.OrderSummaryItem--removable {
  display: block;
}

.OrderSummaryItem__Description {
  display: flex;
  margin-top: 10px;
}

.OrderSummaryItem__Label {
  flex: 1;
  color: #737373;
}

.OrderSummaryItem__Amount {
  display: flex;
  color: #3A3A3A;
  align-items: center;
}

.OrderSummaryItem__Amount .stx-price__amount {
  font-size: inherit;
  color: inherit;
}

.Breakdown__Section {
  gap: 10px;
  display: flex;
  flex-direction: column;
  padding: 20px 0;
  border-bottom: 1px solid rgba(0,0,0,0.1);
}

.Breakdown__Section.Breakdown__Section--Total .OrderSummaryItem__Title {
  font-size: 16px;
}

.Breakdown__Section.Breakdown__Section--Total .OrderSummaryItem__Amount {
  font-size: 24px;
}

.OrderSummaryItem__Action {
  border: none;
  background: none;
  height: auto;
  display: flex;
  align-items: center;
  cursor: pointer;
  justify-content: center;
  margin-right: 5px;
}
