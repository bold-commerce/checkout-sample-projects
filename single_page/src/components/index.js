export * from './Address';
export * from './BillingAddress';
export * from './CheckoutButton';
export * from './CheckoutSection';
export * from './Customer';
export * from './DiscountForm';
export * from './EmptyState';
export * from './Header';
export * from './LineItems';
export * from './LoadingState';
export * from './OrderErrors';
export * from './OrderSummary';
export * from './PaymentMethod';
export * from './ShippingAddress';
export * from './ShippingLines';
