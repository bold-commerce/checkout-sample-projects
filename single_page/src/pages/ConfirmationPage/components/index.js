export * from './ConfirmationList';
export * from './ConfirmationListItem';
