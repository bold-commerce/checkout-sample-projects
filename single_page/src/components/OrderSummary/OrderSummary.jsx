import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { CheckoutSection } from '../CheckoutSection';
import { LineItems } from '../LineItems';
import { DiscountForm } from '../DiscountForm';
import { OrderSummaryBreakdown, OrderSummaryCollapseButton } from '.';
import './OrderSummary.css';
import { useTranslation } from 'react-i18next';

const OrderSummary = ({ readOnly, summaryOpen, onCollapse }) => {
  const { t } = useTranslation();
  return (
    <>
      <OrderSummaryCollapseButton onClick={onCollapse} summaryOpen={summaryOpen} />
      <div id="OrderSummary" className="OrderSummary">
        <CheckoutSection
          className="FieldSet--OrderSummary"
          title={t('order_summary')}
        >
          <LineItems readOnly={readOnly} />
          { !readOnly && <DiscountForm /> }
          <OrderSummaryBreakdown />
        </CheckoutSection>
      </div>
    </>
  );
};

OrderSummary.propTypes = {
  readOnly: PropTypes.bool,
};

export default OrderSummary;
