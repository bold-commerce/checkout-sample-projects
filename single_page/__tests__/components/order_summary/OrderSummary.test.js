import React from "react";
import { render } from "@testing-library/react";
import { OrderSummary } from "../../../client/components/OrderSummary";
import { exampleLineItems as MOCKlineItems } from '../../utils/lineItemHelpers';
import { exampleBreakdown as MOCKbreakdown } from '../../utils/orderHelpers';

jest.mock('@boldcommerce/checkout-react-components', () => ({
  useLineItems: () => ({
    data: MOCKlineItems,
    updateLineItemQuantity: (() => {}),
    removeLineItem: (() => {})
  }),
  useDiscount: () => ({
    data: {
      discountTotal: 123,
      discountApplied: true,
      discountCode: 'Test'
    },
    applyDiscount: (() => {})
  }),
  useBreakdown: () => ({
    data: MOCKbreakdown
  })
}))

describe('OrderSummary', () => {
  test('renders OrderSummary component', () => {
    const { asFragment } = render( <OrderSummary /> )

    expect(asFragment()).toMatchSnapshot();
  })
});
