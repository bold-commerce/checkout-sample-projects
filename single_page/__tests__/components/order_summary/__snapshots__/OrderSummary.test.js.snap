// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderSummary renders OrderSummary component 1`] = `
<DocumentFragment>
  <button
    aria-controls="OrderSummary"
    class="CollapseButton"
    type="button"
  >
    <span
      class="CollapseButton__title"
    >
      <svg
        fill="none"
        height="13"
        viewBox="0 0 17 13"
        width="17"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.2153 11.9455C15.5183 12.338 16.0448 12.3778 16.3912 12.0344C16.7375 11.6909 16.7727 11.0942 16.4696 10.7017L8.96544 0.711341C8.63343 0.281291 8.04315 0.281276 7.71113 0.711309L0.206129 10.7016C-0.0969428 11.0942 -0.0618533 11.6908 0.284508 12.0343C0.630869 12.3778 1.15734 12.338 1.46041 11.9455L8.33824 2.76748L15.2153 11.9455Z"
          fill="#A1A1A1"
        />
      </svg>
      View order summary
    </span>
    <span
      class="CollapseButton__description"
    >
      (6 items)
      <div
        class="stx-price stx-price--alight-center"
      >
        <p
          aria-label="Product price"
          class="stx-price__amount"
        >
          $32.40
        </p>
      </div>
    </span>
  </button>
  <div
    class="OrderSummary"
    id="OrderSummary"
  >
    <section
      class="FieldSet FieldSet--OrderSummary"
    >
      <div
        class="FieldSet__Header"
      >
        <h3
          class="FieldSet__Heading"
        >
          Order Summary
        </h3>
      </div>
      <div
        class="FieldSet__Content"
      >
        <div
          class="LineItemList"
        >
          <div
            class="LineItem"
          >
            <div
              class="LineItem__ImageDescriptionWrapper"
            >
              <div
                class="LineItem__ProductImageWrapper"
              >
                <div
                  class="stx-image"
                >
                  <div
                    class="stx-image--content"
                  >
                    <div
                      class="stx-image--holder"
                    >
                      <img
                        alt="Size Large / Colour Blue"
                        class="stx-image--image"
                        src=""
                        title="Size Large / Colour Blue"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="LineItem__ProductDetails"
              >
                <div
                  class="stx-details stx-details--left"
                >
                  <div
                    class="stx-details__title"
                  >
                    Size Large / Colour Blue
                  </div>
                  <div
                    class="stx-details__description"
                  />
                </div>
                <button
                  aria-busy="false"
                  aria-disabled="false"
                  class="stx-button stx-button--secondary"
                >
                  <span
                    class="stx-button__text"
                  >
                    Remove
                  </span>
                  <span
                    class="stx-loading-spinner stx-button__spinner"
                  />
                </button>
              </div>
            </div>
            <div
              class="LineItem__QuantityPriceWrapper"
            >
              <div
                class="LineItem__LineItemQuantityWrapper"
              >
                <div
                  class="LineItemQuantity__Decrement"
                >
                  <button
                    aria-busy="false"
                    aria-disabled="true"
                    class="stx-button stx-button--is-disabled"
                    disabled=""
                  >
                    <span
                      class="stx-button__text"
                    >
                      -
                    </span>
                    <span
                      class="stx-loading-spinner stx-button__spinner"
                    />
                  </button>
                </div>
                <div
                  class="LineItemQuantity"
                >
                  <div
                    class="stx-input"
                  >
                    <input
                      aria-readonly="false"
                      class="stx-input__input-element"
                      type="number"
                      value="1"
                    />
                  </div>
                </div>
                <div
                  class="LineItemQuantity__Increment"
                >
                  <button
                    aria-busy="false"
                    aria-disabled="false"
                    class="stx-button"
                  >
                    <span
                      class="stx-button__text"
                    >
                      +
                    </span>
                    <span
                      class="stx-loading-spinner stx-button__spinner"
                    />
                  </button>
                </div>
              </div>
              <div
                class="LineItem__ProductPrice"
              >
                <div
                  class="stx-price stx-price--alight-center"
                >
                  <p
                    aria-label="Product price"
                    class="stx-price__amount"
                  >
                    $14.00
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="LineItem"
          >
            <div
              class="LineItem__ImageDescriptionWrapper"
            >
              <div
                class="LineItem__ProductImageWrapper"
              >
                <div
                  class="stx-image"
                >
                  <div
                    class="stx-image--content"
                  >
                    <div
                      class="stx-image--holder"
                    >
                      <img
                        alt="Number Two"
                        class="stx-image--image"
                        src=""
                        title="Number Two"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="LineItem__ProductDetails"
              >
                <div
                  class="stx-details stx-details--left"
                >
                  <div
                    class="stx-details__title"
                  >
                    Number Two
                  </div>
                  <div
                    class="stx-details__description"
                  />
                </div>
                <button
                  aria-busy="false"
                  aria-disabled="false"
                  class="stx-button stx-button--secondary"
                >
                  <span
                    class="stx-button__text"
                  >
                    Remove
                  </span>
                  <span
                    class="stx-loading-spinner stx-button__spinner"
                  />
                </button>
              </div>
            </div>
            <div
              class="LineItem__QuantityPriceWrapper"
            >
              <div
                class="LineItem__LineItemQuantityWrapper"
              >
                <div
                  class="LineItemQuantity__Decrement"
                >
                  <button
                    aria-busy="false"
                    aria-disabled="false"
                    class="stx-button"
                  >
                    <span
                      class="stx-button__text"
                    >
                      -
                    </span>
                    <span
                      class="stx-loading-spinner stx-button__spinner"
                    />
                  </button>
                </div>
                <div
                  class="LineItemQuantity"
                >
                  <div
                    class="stx-input"
                  >
                    <input
                      aria-readonly="false"
                      class="stx-input__input-element"
                      type="number"
                      value="2"
                    />
                  </div>
                </div>
                <div
                  class="LineItemQuantity__Increment"
                >
                  <button
                    aria-busy="false"
                    aria-disabled="false"
                    class="stx-button"
                  >
                    <span
                      class="stx-button__text"
                    >
                      +
                    </span>
                    <span
                      class="stx-loading-spinner stx-button__spinner"
                    />
                  </button>
                </div>
              </div>
              <div
                class="LineItem__ProductPrice"
              >
                <div
                  class="stx-price stx-price--alight-center"
                >
                  <p
                    aria-label="Product price"
                    class="stx-price__amount"
                  >
                    $32.00
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="LineItem"
          >
            <div
              class="LineItem__ImageDescriptionWrapper"
            >
              <div
                class="LineItem__ProductImageWrapper"
              >
                <div
                  class="stx-image"
                >
                  <div
                    class="stx-image--content"
                  >
                    <div
                      class="stx-image--holder"
                    >
                      <img
                        alt="Default Title"
                        class="stx-image--image"
                        src=""
                        title="Default Title"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="LineItem__ProductDetails"
              >
                <div
                  class="stx-details stx-details--left"
                >
                  <div
                    class="stx-details__title"
                  >
                    Default Title
                  </div>
                  <div
                    class="stx-details__description"
                  />
                </div>
                <button
                  aria-busy="false"
                  aria-disabled="false"
                  class="stx-button stx-button--secondary"
                >
                  <span
                    class="stx-button__text"
                  >
                    Remove
                  </span>
                  <span
                    class="stx-loading-spinner stx-button__spinner"
                  />
                </button>
              </div>
            </div>
            <div
              class="LineItem__QuantityPriceWrapper"
            >
              <div
                class="LineItem__LineItemQuantityWrapper"
              >
                <div
                  class="LineItemQuantity__Decrement"
                >
                  <button
                    aria-busy="false"
                    aria-disabled="false"
                    class="stx-button"
                  >
                    <span
                      class="stx-button__text"
                    >
                      -
                    </span>
                    <span
                      class="stx-loading-spinner stx-button__spinner"
                    />
                  </button>
                </div>
                <div
                  class="LineItemQuantity"
                >
                  <div
                    class="stx-input"
                  >
                    <input
                      aria-readonly="false"
                      class="stx-input__input-element"
                      type="number"
                      value="3"
                    />
                  </div>
                </div>
                <div
                  class="LineItemQuantity__Increment"
                >
                  <button
                    aria-busy="false"
                    aria-disabled="false"
                    class="stx-button"
                  >
                    <span
                      class="stx-button__text"
                    >
                      +
                    </span>
                    <span
                      class="stx-loading-spinner stx-button__spinner"
                    />
                  </button>
                </div>
              </div>
              <div
                class="LineItem__ProductPrice"
              >
                <div
                  class="stx-price stx-price--alight-center"
                >
                  <p
                    aria-label="Product price"
                    class="stx-price__amount"
                  >
                    $123.00
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="SummaryBlock Summary__DiscountForm"
        >
          <div
            class="DiscountForm"
          >
            <div
              class="stx-field stx-field--is-disabled stx-field--with-input"
            >
              <div
                class="stx-input stx-input--is-disabled"
              >
                <input
                  aria-disabled="true"
                  class="stx-input__input-element"
                  disabled=""
                  id="stxField1"
                  placeholder="Enter discount code"
                  type="text"
                  value="Test"
                />
              </div>
            </div>
            <button
              aria-busy="false"
              aria-disabled="true"
              class="stx-button stx-button--primary stx-button--is-disabled"
              disabled=""
            >
              <span
                class="stx-button__text"
              >
                Apply
              </span>
              <span
                class="stx-loading-spinner stx-button__spinner"
              />
            </button>
          </div>
        </div>
        <div
          class="OrderSummary__Breakdown"
        >
          <div
            class="Breakdown__Section"
          >
            <div
              class="OrderSummaryItem"
            >
              <span
                class="OrderSummaryItem__Title"
              >
                Subtotal
              </span>
              <span
                class="OrderSummaryItem__Amount"
              >
                (
                <div
                  class="stx-price stx-price--alight-center"
                >
                  <p
                    aria-label="Product price"
                    class="stx-price__amount"
                  >
                    $19.19
                  </p>
                </div>
                )
              </span>
            </div>
            <div
              class="OrderSummaryItem OrderSummaryItem--removable"
            >
              <span
                class="OrderSummaryItem__Title"
              >
                Discount
              </span>
              <div
                class="OrderSummaryItem__Description"
              >
                <button
                  class="OrderSummaryItem__Action"
                  type="button"
                >
                  <svg
                    fill="none"
                    height="12"
                    viewBox="0 0 13 12"
                    width="13"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M6.14551 12C9.45922 12 12.1455 9.31371 12.1455 6C12.1455 2.68629 9.45922 0 6.14551 0C2.8318 0 0.145508 2.68629 0.145508 6C0.145508 9.31371 2.8318 12 6.14551 12ZM9.30831 2.80811C9.50357 3.00338 9.50357 3.31996 9.30831 3.51522L6.83789 5.98564L9.32005 8.4678C9.51531 8.66306 9.51531 8.97964 9.32005 9.17491C9.12479 9.37017 8.80821 9.37017 8.61294 9.17491L6.13078 6.69275L3.65707 9.16646C3.4618 9.36172 3.14522 9.36172 2.94996 9.16646C2.7547 8.9712 2.7547 8.65462 2.94996 8.45936L5.42368 5.98564L2.95104 3.513C2.75577 3.31774 2.75577 3.00116 2.95104 2.80589C3.1463 2.61063 3.46288 2.61063 3.65814 2.80589L6.13078 5.27853L8.6012 2.80811C8.79646 2.61285 9.11305 2.61285 9.30831 2.80811Z"
                      fill="#666666"
                      fill-rule="evenodd"
                    />
                  </svg>
                </button>
                <span
                  class="OrderSummaryItem__Label"
                >
                  Discount code: Test
                </span>
                <span
                  class="OrderSummaryItem__Amount"
                >
                  (
                  <div
                    class="stx-price stx-price--alight-center"
                  >
                    <p
                      aria-label="Product price"
                      class="stx-price__amount"
                    >
                      -$1.23
                    </p>
                  </div>
                  )
                </span>
              </div>
            </div>
            <div
              class="OrderSummaryItem"
            >
              <span
                class="OrderSummaryItem__Title"
              >
                Shipping
              </span>
              <span
                class="OrderSummaryItem__Amount"
              >
                (
                <div
                  class="stx-price stx-price--alight-center"
                >
                  <p
                    aria-label="Product price"
                    class="stx-price__amount"
                  >
                    $2.10
                  </p>
                </div>
                )
              </span>
            </div>
            <div
              class="OrderSummaryItem"
            >
              <span
                class="OrderSummaryItem__Title"
              >
                Taxes
              </span>
              <span
                class="OrderSummaryItem__Amount"
              >
                (
                <div
                  class="stx-price stx-price--alight-center"
                >
                  <p
                    aria-label="Product price"
                    class="stx-price__amount"
                  >
                    $11.11
                  </p>
                </div>
                )
              </span>
            </div>
          </div>
          <div
            class="Breakdown__Section Breakdown__Section--Total"
          >
            <div
              class="OrderSummaryItem"
            >
              <span
                class="OrderSummaryItem__Title"
              >
                Total
              </span>
              <span
                class="OrderSummaryItem__Amount"
              >
                (
                <div
                  class="stx-price stx-price--alight-center"
                >
                  <p
                    aria-label="Product price"
                    class="stx-price__amount"
                  >
                    $32.40
                  </p>
                </div>
                )
              </span>
            </div>
          </div>
          <div
            class="Breakdown__Section"
          >
            <div
              class="OrderSummaryItem OrderSummaryItem--removable"
            >
              <span
                class="OrderSummaryItem__Title"
              >
                Payments
              </span>
              <div
                class="OrderSummaryItem__Description"
              >
                <span
                  class="OrderSummaryItem__Label"
                >
                  <span
                    class="RedactedCard"
                  >
                    <span
                      class="RedactedCard__Brand"
                    >
                      visa
                    </span>
                    <span
                      class="RedactedCard__Number"
                    >
                       •••• •••• •••• test card
                    </span>
                  </span>
                </span>
                <span
                  class="OrderSummaryItem__Amount"
                >
                  (
                  <div
                    class="stx-price stx-price--alight-center"
                  >
                    <p
                      aria-label="Product price"
                      class="stx-price__amount"
                    >
                      $1.00
                    </p>
                  </div>
                  )
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
